system/jlib/jlog.hpp

```diff
@@ -1342,4 +1343,295 @@ interface IComponentLogFileCreator : extends IInterface
+    virtual void toString(StringBuffer & out) const = 0;
+    virtual LogAccessFilterType filterType() const = 0;
+};
+
+class CLogAccessFilter : implements CInterfaceOf<ILogAccessFilter> {};
+
+extern jlib_decl ILogAccessFilter * getLogAccessFilterFromPTree(IPropertyTree * tree);
+extern jlib_decl ILogAccessFilter * getJobIDLogAccessFilter(const char * wuid);
+extern jlib_decl ILogAccessFilter * getComponentLogAccessFilter(const char * component);
+extern jlib_decl ILogAccessFilter * getAudienceLogAccessFilter(MessageAudience audience);
+extern jlib_decl ILogAccessFilter * getClassLogAccessFilter(LogMsgClass logclass);
+extern jlib_decl ILogAccessFilter * getBinaryLogAccessFilter(ILogAccessFilter * arg1, ILogAccessFilter * arg2, LogAccessFilterType type);
+extern jlib_decl ILogAccessFilter * getBinaryLogAccessFilterOwn(ILogAccessFilter * arg1, ILogAccessFilter * arg2, LogAccessFilterType type);
+extern jlib_decl ILogAccessFilter * getWildCardLogAccessFilter();
+
+struct LogAccessConditions
+{
+private:
+    Owned<ILogAccessFilter> filter;
+public:
+    LogAccessTimeRange timeRange;
+    unsigned limit = 100;
+    long offset = 0;
+
+    StringArray logFieldNames;
+
+    LogAccessConditions & operator = (const LogAccessConditions & l)
+    {
+        copyLogFieldNames(&l.logFieldNames);
+        limit = l.limit;
+        timeRange = l.timeRange;
+        filter = l.filter;
+        offset = l.offset;
+
+        return *this;
+    }
+
+    ~LogAccessConditions() {}
+
+    ILogAccessFilter * queryFilter()
+    {
+        return filter.get();
+    }
+    void setFilter(ILogAccessFilter * _filter)
+    {
+        filter.setown(_filter);
+    }
+
+    void appendLogFieldName(const char * fieldname)
+    {
+        if (!logFieldNames.contains(fieldname))
+            logFieldNames.append(fieldname);
+    }
+
+    void copyLogFieldNames(const StringArray * fields)
+    {
+        if (*fields)
+        {
+            ForEachItemIn(fieldsindex,*fields)
+            {
+                appendLogFieldName(fields->item(fieldsindex));
+            }
+        }
+    }
+    inline const StringArray &queryLogFieldNames() { return logFieldNames; }
+};
+
+typedef enum
+{
+    LOGACCESS_LOGFORMAT_xml,
+    LOGACCESS_LOGFORMAT_json,
+    LOGACCESS_LOGFORMAT_csv
+} LogAccessLogFormat;
+
+inline LogAccessLogFormat logAccessFormatFromName(char const * name)
+{
+    if (isEmptyString(name))
+        throw makeStringException(-1, "Encountered empty Log Access Format name");
+
+    if(strieq(name, "xml"))
+        return LOGACCESS_LOGFORMAT_xml;
+    else if(strieq(name, "json"))
+        return LOGACCESS_LOGFORMAT_json;
+    else if(strieq(name, "csv"))
+        return LOGACCESS_LOGFORMAT_csv;
+    else
+        throw makeStringExceptionV(-1, "Encountered unknown Log Access Format name: '%s'", name);
+}
+
+interface IRemoteLogAccess : extends IInterface
+{
+    virtual bool fetchLog(LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format) = 0;
```

Comment:
trivial: should it be: const LogAccessConditions &option ? (if fetchLog should never change options)